test/tests/bootstrap/test_api20_windows_bootstrap.py

```diff
@@ -0,0 +1,137 @@
+           "options": {"defaults": {"version": "2012",
+                                    "repo": "http://172.31.128.1:8080/repo/winpe",
+                                    "smbRepo": "\\\\172.31.128.1\\windowsServer2012",
+                                    "productkey": "XXXXX-XXXXX-XXXXX-XXXXX-XXXXX",
+                                    "username": "rackhduser",
+                                    "password": "RackHDRocks",
+                                    "smbUser": "vagrant",
+                                    "smbPassword": "vagrant"}}}
+}
+
+Example command line using external payload file:
+
+python run_tests.py -stack 4 -test tests/bootstrap/test_api20_windows_bootstrap.py -extra base_windows_2012_install.json
+
+RackHD Windows installation workflow requires special configuration of the RackHD server:
+- A customized WinPE environment installed on RackHD server as documented here:
+        https://github.com/RackHD/on-tools/tree/master/winpe
+- Samba installed on the RackHD server and configured as documented here:
+        http://rackhd.readthedocs.io/en/latest/rackhd/install_os.html?highlight=os%20install
+- Windows 2012 installation distro installed on RackHD server or equivalent NFS mount.
+- Windows 2012 activation key in the installation payload file.
+
+'''
+
+import fit_path  # NOQA: unused import
+from nose.plugins.attrib import attr
+import fit_common
+import flogging
+import random
+import time
+from nosedep import depends
+log = flogging.get_loggers()
+
+# sample default base payload
+PAYLOAD = {"name": "Graph.InstallWindowsServer",
+           "options": {"defaults": {"version": "2012",
+                                    "repo": "http://172.31.128.1:8080/repo/winpe",
+                                    "smbRepo": "\\\\172.31.128.1\\windowsServer2012",
+                                    "productkey": "XXXXX-XXXXX-XXXXX-XXXXX-XXXXX",
+                                    "username": "rackhduser",
+                                    "password": "RackHDRocks",
+                                    "smbUser": "vagrant",
+                                    "smbPassword": "vagrant"}}}
+# if an external payload file is specified, use that
+config = fit_common.fitcfg().get('bootstrap-payload', None)
+if config:
+    PAYLOAD = config
+
+
+# this routine polls a workflow task ID for completion
+def wait_for_workflow_complete(instanceid, start_time, waittime=3200, cycle=30):
+    log.info_5(" Workflow started at time: " + str(start_time))
+    while time.time() - start_time < waittime:  # limit test to waittime seconds
+        result = fit_common.rackhdapi("/api/2.0/workflows/" + instanceid)
+        if result['status'] != 200:
+            log.error(" HTTP error: " + result['text'])
+            return False
+        if result['json']['status'] in ['running', 'pending']:
+            log.info_5("{} workflow status: {}".format(result['json']['injectableName'], result['json']['status']))
```

Comment:
The comment that Larry made in the earlier PR.  Flogger messages should not use format or " + ". Standard "C" formatting is to be used.  Rather than propogating further, should fix it here.